src/builtins/core/datetime.rs

```diff
@@ -682,8 +683,33 @@ impl PlainDateTime {
         Ok(Self::new_unchecked(result, self.calendar.clone()))
     }
 
+    pub fn to_zoned_date_time_with_provider(
```

Comment:
nit: add a `PlainDateTime::to_zoned_date_time` implementation on `PlainDateTime` in compiled.

It looks like I missed adding this stub. So a new module should be added to the compiled directory for `PlainDateTime` and the companion method added.

Worthwhile note: there's no current need to add this method to `temporal_capi` just yet because time zone providers over the FFI is still a work in progress. So only the non provider method implementation is needed.